src/rma/js/components/destruction-list/destruction-form.js

```diff
@@ -0,0 +1,134 @@
+import React, {useEffect, useState} from "react";
+
+import { SelectMultipleInput } from "./select";
+import { DateInput } from "./inputs";
+import { ZakenTable } from "./zaken-table";
+import { CreateModal} from "./create-modal";
+import { countObjectKeys } from "../../utils";
+
+
+function getFullUrl(url, filters) {
+    const query = Object.keys(filters).filter(k=>filters[k]).map(k => `${k}=${filters[k]}`).join('&');
+    return `${url}?${query}`;
+}
+
+
+function DestructionForm(props) {
+    const { zaaktypen, reviewers, zakenUrl, url, csrftoken } = props;
+
+    //filters
+    const [selectedZaaktypen, setSelectedZaaktypen] = useState([]);
+    const [selectedStartdatum, setSelectedStartdatum] = useState(null);
+    const filters = {
+        zaaktypen: selectedZaaktypen,
+        startdatum: selectedStartdatum,
+    };
+
+    //load zaken
+    const [error, setError] = useState(null);
+    const [isLoaded, setIsLoaded] = useState(false);
+    const [zaken, setZaken] = useState([]);
+
+    // checkboxes
+    const [checkboxes, setCheckboxes] = useState({});
```

Comment:
this quickly gets messy, because you're tracking different states which is actually one big state.

A refactor (later) should merge all this together and make use of `useReducer` probably.

Reply:
yep, later :tm: